feat: adds release mode docs to repack doc (#108)
ZephyrCloudIO/zephyr-documentation: components/zephyr/repack/start-from-scratch.mdx

<Image src="/host-app-ios.webp" alt="Zephyr Cloud Minimal example running in iOS emulator" />

The process should be the same for Android. 

## Running the application in release mode with Zephyr

> By default, the HostApp loads miniapp bundles from a local development server (e.g. localhost). To run the app in production/release mode without using the Zephyr development environment, you must replace those local URLs with the Zephyr Cloud URLs for each miniapp. This ensures the HostApp fetches the miniapp bundles directly from the Zephyr-hosted endpoints. The process involves configuring your Zephyr environment, retrieving the correct remote URLs, and updating the HostApp’s module federation configuration. Once completed, you can build and launch the app in release mode normally (e.g. via Xcode or react-native release build), and it will load miniapps from the Zephyr Cloud URLs instead of localhost.

Reviewer: ```suggestion
By default, the HostApp loads miniapp bundles from a local development server (e.g. localhost). To run the app in production/release mode without using the Zephyr development environment, you must replace those local URLs with the Zephyr Cloud URLs for each miniapp. This ensures the HostApp fetches the miniapp bundles directly from the Zephyr-hosted endpoints. The process involves configuring your Zephyr environment - and zephyr will auto-retrieving the correct remote URLs as well as updating the HostApp’s module federation configuration. Once completed, you can build and launch the app in release mode normally (e.g. via Xcode or react-native release build), and it will load miniapps from the Zephyr Cloud URLs instead of localhost.
```

> To successfully run the HostApp in release mode using Zephyr-hosted miniapps, you need to have a Zephyr environment configured for your project (such as production or staging). Each miniapp (remote module), along with the HostApp itself, must be registered within this environment. For step-by-step setup guidance, refer to the [Zephyr Environment Guide](../../how-to/environments.mdx)

Reviewer: ```suggestion
To successfully run the HostApp in release mode using Zephyr-hosted MiniApps, you need to have a Zephyr environment configured for your project (such as production or staging). Each MiniApps (remote module), along with the HostApp itself, must be registered within this environment. For step-by-step setup guidance, refer to the [Zephyr Environment Guide](../../how-to/environments.mdx)
```

### Steps

Follow these steps to configure and run the HostApp in release mode with Zephyr:

Configure the Zephyr environment:

> 1. Change your configs to divide debug builds and release builds:

Reviewer: ```suggestion
1. Change your configs to divide debug builds and release builds:
```
How's debug builds and release builds separated here?

   "scripts": {
      "bundle": "pnpm run bundle:ios && pnpm run bundle:android",
      "bundle:ios": "react-native bundle --platform ios --dev false --entry-file index.js",
      "bundle:android": "react-native bundle --platform android --dev false --entry-file index.js"
    }
   ```

   ```json title="MiniApp/package.json"
   "scripts": {
    "bundle": "pnpm run bundle:ios && pnpm run bundle:android",
    "bundle:ios": "react-native bundle --platform ios --dev false --entry-file index.js",
    "bundle:android": "react-native bundle --platform android --dev false --entry-file index.js"
   }
   ```

2. Bundle HostApp and MiniApps, and upload to Zephyr Cloud:

  - Bundle the MiniApp:

   ```bash 
   ZC=1 pnpm --filter MiniApp bundle
   ```
   
   - Bundle the HostApp:

   ```bash 
   ZC=1 pnpm --filter HostApp bundle
   ```

   - Command run with `ZC=1` will upload the bundles to Zephyr Cloud.

3. Build and run the HostApp in release mode:

   - For Android, you can use the following command from the HostApp android directory:

   ```bash
   ZC=1 ./gradlew assembleRelease
   ```

   {/* TODO: set ZC for iOS */}
   {/* - For iOS, you can use the following command: */}

>    - The HostApp will now load each miniapp bundle from the specified Zephyr Cloud URLs instead of localhost. Verify that the app launches correctly and that each miniapp is fetched successfully from the remote URL.

Reviewer: ```suggestion
   - The HostApp will now load each miniapp bundle from the specified Zephyr Cloud URLs instead of localhost. Verify that the app launches correctly and that each MiniApp is fetched successfully from the remote URL.
```

> By following these steps, you can run your HostApp in production mode without the Zephyr development environment, using the remote module bundles hosted on Zephyr Cloud.

Reviewer: ```suggestion
By following these steps, you can run your HostApp in production mode without the Zephyr development environment, using the remote module bundles hosted on Zephyr Cloud.
```

so what's the separation between "with" and "without" zephyr development environment?

Author: It's changed to 

> By following these steps, you can run your HostApp in production mode with the Zephyr, using the remote bundles hosted on Zephyr Cloud.

<Image src="/host-app-ios.webp" alt="Zephyr Cloud Minimal example running in iOS emulator" />

The process should be the same for Android. 

## Running the application in release mode with Zephyr

> By default, the HostApp loads miniapp bundles from a local development server (e.g. localhost). To run the app in production/release mode without using the Zephyr development environment, you must replace those local URLs with the Zephyr Cloud URLs for each miniapp. This ensures the HostApp fetches the miniapp bundles directly from the Zephyr-hosted endpoints. The process involves configuring your Zephyr environment - and zephyr will auto-retrieving the correct remote URLs as well as updating the HostApp’s module federation configuration. Once completed, you can build and launch the app in release mode normally (e.g. via Xcode or react-native release build), and it will load miniapps from the Zephyr Cloud URLs instead of localhost.

Reviewer: Or move the paragraph at line 545 to the top, have a <h3> tittle: 
1. release with zephyr 
2. release without zephyr (note it will break runtime sdk)

To successfully run the HostApp in release mode using Zephyr-hosted MiniApps, you need to have a Zephyr environment configured for your project (such as production or staging). Each MiniApps (remote module), along with the HostApp itself, must be registered within this environment. For step-by-step setup guidance, refer to the [Zephyr Environment Guide](../../how-to/environments.mdx)

### Steps

Follow these steps to configure and run the HostApp in release mode with Zephyr:

Configure the Zephyr environment:

1. Change your configs to divide debug builds and release builds:

   - In the HostApp and MiniApp, change the `rspack.config.mjs` file to add a `ZC` variable:

> 

Reviewer: ```suggestion
#### HostApp's configuration
```

            new Repack.RepackPlugin({
              platform,
            }),
            new Repack.plugins.ModuleFederationPluginV2({
              name: 'HostApp',
              filename: 'HostApp.container.js.bundle',
              dts: false,
              remotes: {
                MiniApp: `MiniApp@http://localhost:9001/${platform}/MiniApp.container.js.bundle`,
              },
              shared: {
                react: {
                  singleton: true,
                  version: '19.0.0',
                  eager: true,
                },
                'react-native': {
                  singleton: true,
                  version: '0.78.0',
                  eager: true,
                }
              },
           }),
          // Supports for new architecture - Hermes can also use JS, it's not a requirement, it will still work the same but it's for performance optimization
          new Repack.plugins.HermesBytecodePlugin({
            enabled: mode === 'production',
            test: /\.(js)?bundle$/,
            exclude: /index.bundle$/,
            }),
          ],
        }
      };

      export default USE_ZEPHYR ? withZephyr()(config) : config;
   ```

> 

Reviewer: ```suggestion
#### MiniApp's configuration
```

        },
        module: {
          rules: [
            ...Repack.getJsTransformRules(),
            ...Repack.getAssetTransformRules({inline: true}),
          ],
        },
        plugins: [
          new Repack.RepackPlugin(),
          new Repack.plugins.ModuleFederationPluginV2({
            name: 'MiniApp',
            filename: 'MiniApp.container.js.bundle',
            dts: false,
            exposes: {
              './App': './App.tsx',
            },
            shared: {
              react: {
                singleton: true,
                version: '19.0.0',
                eager: STANDALONE,
              },
              'react-native': {
                singleton: true,
                version: '0.78.0',
                eager: STANDALONE,
              },
            },
          }),
          new Repack.plugins.HermesBytecodePlugin({
            enabled: mode === 'production',
            test: /\.(js)?bundle$/,
            exclude: /index.bundle$/,
          }),
          ],
        };
      };

    export default USE_ZEPHYR ? withZephyr()(config) : config;

    ```

    - `ZC` is used to indicate that the bundles are for Zephyr Cloud.

>     - Now, when you run or bundle with `ZC=1`, the bundles will be uploaded to Zephyr Cloud.

Reviewer: ```suggestion
    - Now, when you run or bundle with `ZC=1`, the bundles will be deployed to Zephyr Cloud.
```

2. Add bundle scripts to the HostApp and MiniApp:

   - In the HostApp and MiniApp, add the following scripts to the package.json files:

   ```json title="HostApp/package.json"
   "scripts": {
      "bundle": "pnpm run bundle:ios && pnpm run bundle:android",
      "bundle:ios": "react-native bundle --platform ios --dev false --entry-file index.js",
      "bundle:android": "react-native bundle --platform android --dev false --entry-file index.js"
    }
   ```

   ```json title="MiniApp/package.json"
   "scripts": {
    "bundle": "pnpm run bundle:ios && pnpm run bundle:android",
    "bundle:ios": "react-native bundle --platform ios --dev false --entry-file index.js",
    "bundle:android": "react-native bundle --platform android --dev false --entry-file index.js"
   }
   ```

> 2. Bundle HostApp and MiniApps, and upload to Zephyr Cloud:

Reviewer: ```suggestion
2. Bundle HostApp and MiniApps, and deploy to Zephyr Cloud:
```

  - Bundle the MiniApp:

   ```bash 
   ZC=1 pnpm --filter MiniApp bundle
   ```
   
   - Bundle the HostApp:

   ```bash 
   ZC=1 pnpm --filter HostApp bundle
   ```

   - Command run with `ZC=1` will upload the bundles to Zephyr Cloud.

3. Build and run the HostApp in release mode:

   - For Android, you can use the following command from the HostApp android directory:

   ```bash
   ZC=1 ./gradlew assembleRelease
   ```

   {/* TODO: set ZC for iOS */}

>    {/* - For iOS, you can use the following command: */}

Reviewer: god we need to fix this

>    - The HostApp will now load each miniapp bundle from the specified Zephyr Cloud URLs instead of localhost. Verify that the app launches correctly and that each MiniApp is fetched successfully from the remote URL.

Reviewer: ```suggestion
   - The HostApp will now load each MiniApp's bundle from the specified Zephyr Cloud URLs instead of localhost. Verify that the app launches correctly and that each MiniApp is fetched successfully from the remote URL.
```

> By following these steps, you can run your HostApp in production mode with the Zephyr, using the remote bundles hosted on Zephyr Cloud.

Reviewer: ```suggestion
By following these steps, you can run your HostApp in production mode with Zephyr, using the remote bundles (MiniApps) deployed on Zephyr Cloud.
```

<Image src="/host-app-ios.webp" alt="Zephyr Cloud Minimal example running in iOS emulator" />

The process should be the same for Android. 

> ## Running the application in release mode without Zephyr

Reviewer: if they don't use zephyr in release mode there won't be global environment variable injected for runtime SDK - they must use it during bundling. 

the correct mini app's URL would be resolved or replaced either by: 
1. zephyr-repack-plugin 
2. runtime SDK + zephyr-repack-plugin + repack `ScriptManager`

To successfully run the HostApp in release mode using Zephyr-hosted miniapps, you need to have a Zephyr environment configured for your project (such as production or staging). Each miniapp (remote module), along with the HostApp itself, must be registered within this environment. For step-by-step setup guidance, refer to the [Zephyr Environment Guide](../../how-to/environments.mdx)

### Steps

Follow these steps to configure and run the HostApp in release mode with Zephyr:

Configure the Zephyr environment:

1. Configure the Zephyr environment:

   - In the Zephyr Cloud dashboard, create or select the target environment (for example, production or staging).

2. Retrieve the remote URLs:

   - After the environment is configured, open its details page in the Zephyr dashboard for each platform. Zephyr will display the hosted URLs for the HostApp and each miniapp.

   - Copy the provided HostApp URL. It usually follows this pattern:

>      `https://{platform}-{appname}-{yourusername}-ze.zephyrcloud.app/`

Author: Yeah, after the conversion with you, I see the steps should be reworked
